src/containerapp/azext_containerapp/dotnet_component_decorator.py

```diff
@@ -0,0 +1,80 @@
+# coding=utf-8
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from typing import Any, Dict
+
+from azure.cli.core.commands import AzCliCommand
+from azure.cli.core.azclierror import CLIInternalError
+from azure.cli.command_modules.containerapp.base_resource import BaseResource
+from knack.log import get_logger
+
+from ._models import DotNetComponent as DotNetComponentModel
+
+from ._client_factory import handle_raw_exception
+
+logger = get_logger(__name__)
+
+
+class DotNetComponentDecorator(BaseResource):
+    def __init__(self, cmd: AzCliCommand, client: Any, raw_parameters: Dict, models: str):
+        super().__init__(cmd, client, raw_parameters, models)
+        self.dotnet_component_def = DotNetComponentModel
+
+    def get_argument_environment_name(self):
+        return self.get_param("environment_name")
+
+    def get_argument_dotnet_component_name(self):
+        return self.get_param("dotnet_component_name")
+
+    def get_argument_component_type(self):
+        return self.get_param("dotnet_component_type")
+
+    def set_argument_component_type(self, component_type: str):
+        self.set_param("dotnet_component_type", component_type)
+
+    def construct_payload(self):
+        if self.get_argument_component_type() != "AspireDashboard":
+            logger.warning("Supported DotNet component type is: AspireDashboard. Setting component type to AspireDashboard.")
```

Comment:
Should we just override their component type like this? I feel like this could get easily missed by the customer, and it doesn't seem to add a ton of value. I'd prefer just letting our RP handle erroring out here, so we could potentially add new dotnet components in the future?

Reply:
The goal of this is to default to `AspireDashboard` if customer provided another dotnet component type instead of erroring them out.